src/main/java/com/celements/rendering/RenderCommand.java

```diff
@@ -139,23 +141,32 @@ public String renderCelementsDocument(XWikiDocument cellDoc, String renderMode)
 
   public String renderCelementsDocument(XWikiDocument cellDoc, String lang, String renderMode)
       throws XWikiException {
-    LOGGER.debug("renderCelementsDocument: cellDoc [" + cellDoc.getDocumentReference() + "] lang ["
-        + lang + "] renderMode [" + renderMode + "].");
-    String cellDocFN = getWebUtilsService().getRefDefaultSerializer().serialize(
+    LOGGER.debug("renderCelementsDocument: cellDoc [{}] lang [{}] renderMode [{}].",
+        cellDoc.getDocumentReference(), lang, renderMode);
+    String cellDocFN = getModelUtils().serializeRef(
         cellDoc.getDocumentReference());
     if ((getContext() != null) && (getContext().get("vcontext") != null)
         && getContext().getWiki().getRightService().hasAccessLevel(renderMode,
             getContext().getUser(), cellDocFN, getContext())) {
       VelocityContext vcontext = (VelocityContext) getContext().get("vcontext");
       vcontext.put("celldoc", cellDoc.newDocument(getContext()));
-      PageTypeReference cellTypeRef = getPageTypeResolver().getPageTypeRefForDocWithDefault(cellDoc,
-          defaultPageTypeRef);
+      @NotNull
+      Optional<PageTypeReference> optCellTypeRef = getPageTypeResolver()
+          .resolvePageTypeReference(cellDoc);
+      PageTypeReference cellTypeRef;
+      if (defaultPageTypeRef != null) {
+        cellTypeRef = optCellTypeRef.or(defaultPageTypeRef);
+      } else {
+        cellTypeRef = optCellTypeRef.orNull();
+      }
```

Comment:
use java Optional instead of Guava Optional. Also why anotate a local variable?
```suggestion
      PageTypeReference cellTypeRef = getPageTypeResolver()
          .resolvePageTypeReference(cellDoc).toJavaUtil()
          .orElse(defaultPageTypeRef);
```